core/src/main/java/org/everit/json/schema/loader/SchemaLoader.java

```diff
@@ -363,6 +366,20 @@ public SchemaLoader(SchemaLoaderBuilder builder) {
         return builder;
     }
 
+    private ConditionalSchema.Builder buildConditionalSchema() {
+        ConditionalSchema.Builder builder = ConditionalSchema.builder();
+        if (ls.schemaJson().containsKey("if")) {
+            builder.ifSchema(loadChild(ls.schemaJson().require("if").requireObject()).build());
+        }
```

Comment:
The schema is not always an object, it can also be a boolean. so please don't call requireObject() here.
Also, instead of an explicit `if` followed by a `schemaJson().require("if")` call can be replaced by a `maybe("if")` call which returns an`Optional<JsonValue>`. So eg. line 371 - 373 can be replaced with something like:

`ls.schemaJson().maybe("if").map(this::loadChild).ifPresent(builder::ifSchema)`

The same goes for handling `then` and `else`.